src/main/scala/mesosphere/marathon/api/v2/AppsResource.scala

```diff
@@ -323,7 +327,15 @@ class AppsResource @Inject() (
     req: HttpServletRequest, allowCreation: Boolean)(implicit identity: Identity): Future[Response] = async {
     val appId = id.toRootPath
 
-    val appUpdate = canonicalAppUpdateFromJson(appId, body, partialUpdate)
+    val appDoesNotExist = groupManager.app(appId).isEmpty
```

Comment:
There is a race condition here. I believe it is not trivial to get rid of it, and a comment warning about it would suffice.